### ISSUE TITLE
[ADAP-511] [Bug] redaction of row values fails if they contain newline characters

### ISSUE THREAD

**Issue by schielke** [dbt-labs/dbt-snowflake#589]
### Is this a new bug in dbt-snowflake?

- [X] I believe this is a new bug in dbt-snowflake
- [X] I have searched the existing issues, and I could not find an existing issue for this bug

### Current Behavior

I got a "Database Error in model <model_name> (<path_to_model.sql>)\n  100090 (42P18): Duplicate row detected during DML action:  Row Values: ..." (see below) and the row values are not redacted anymore as they were in dbt 1.4. It seems that this is caused by \n characters within the row (in a JSON object) such that the regular expression does not match the string between the brackets.

### Expected Behavior

row values should be redacted

### Steps To Reproduce

dbt run --select <model which causes the duplicate row>

### Relevant log output

```shell
Database Error in model s_telemetry_events_raw (models/stage/s_telemetry_events_raw.sql)\n  100090 (42P18): Duplicate row detected during DML action\n  Row Values: [ \"�\u0000\u0002�\fsignalstatus��\u0001�\n  deviceinfo��\u0001�\bdeviceid�\u00044201�\u0005event�-�+�\u0007eventid�\u0003@97�\n  eventvalue�\u000fFinish job null�\bjobphase�.\u0001�+\u0001�\u0006amount�\u001b\u0004�\fcostcenterid�\u00044201�\u0005jobid�\u000b10.18.2Sch1�\tjobpartid�\u0006SFP0.C�\n  misdetails�\u0014�\bworktype�\bOriginal�\u0004part�#�!�\tsheetname�\bUmschlag�\u0004side�\u0004Back�\u0010percentcompleted�\u0000\u0000d����\r\u0010�\fqueueentryid�\u0000�\tstarttime�\u001d2023-04-20T14:06:47.000+02:00�\u0006status�\tCompleted�\u0005waste�\\�\n  workstepid�\u0018_210311_162114716_042755�\u0011productioncounter�\u001b\u0004�\u0005speed\u0000�\u0006status�\u0004Idle�\rstatusdetails�\u0006Idling�\u0016totalproductioncounter�ǹ\n  ]\n  compiled Code at target/run/idwh/models/stage/s_telemetry_events_raw.sql
```

### Environment

```markdown
- OS: macOS 13.3.1 (a)
- Python: Python 3.11.3
- dbt-core: 1.5.0
- dbt-snowflake: 1.5.0
```

**Comment by dbeatty10** [comment]
Thanks for reporting this @schielke !

I was able to produce the error you described. See below for a simple reproduction case.

### Reprex

1. Create an incremental model

  `models/my_model.sql`
  ```sql
  {{
      config(
          materialized='incremental',
          unique_key='id',
          incremental_strategy='merge',
      )
  }}
  
  -- select 5 as id, to_variant(parse_json('{"key3": "value3\\nleaked"}')) as my_column union all
  select 5 as id, to_variant(parse_json('{"key3": "value3\\nleaked"}')) as my_column
  ```
2. Run it
   ```shell
   dbt run -s my_model
   ```

3. Un-comment the duplicate row in `models/my_model.sql`

4. Run it again
   ```shell
   dbt run -s my_model
   ```

5. Notice that some values are emitted to standard out and logs

**Comment by schielke** [comment]
I created a pull request with a fix: #588
